### PR TITLE
[EdgeDB] Partnership seeds

### PR DESCRIPTION
**Pull request by brentkulwicki**
## Description
Seeding for Partnerships in EdgeDB

- [x] Change the task url above to the actual Monday task
- [x] Add reviewers to this PR

### CODE REVIEW

**Review by willdch — APPROVED** [review]
This looks great to me!

File: dbschema/seeds/017.partnerships.edgeql [hunk 1]
```diff
@@ -0,0 +1,65 @@
+with
+partnershipsJson := to_json($$[
```
Comment (willdch) [inline dbschema/seeds/017.partnerships.edgeql:2]: Just a questions as I have not seen this before.  What is the purpose of the `$$` at the beginning and end of the array?

**Review by CarsonF — COMMENTED** [review]
Looks good! It would be good to have a few projects with multiple partnerships though. And at least one case with multiple _funding_ partners.

File: dbschema/seeds/017.partnerships.edgeql [hunk 2]
```diff
@@ -0,0 +1,65 @@
+with
+partnershipsJson := to_json('[
```
Comment (CarsonF) [inline dbschema/seeds/017.partnerships.edgeql:2]: FWIW I prefer the `$$` for these huge json strings. Easier to not mess up escapes.

Comment (brentkulwicki) [reply dbschema/seeds/017.partnerships.edgeql:2]: Will change back. I just updated to be similar to what everyone else had, but I like $$ too

File: dbschema/seeds/017.partnerships.edgeql [hunk 3]
```diff
@@ -0,0 +1,83 @@
+with
+partnershipsJson := to_json($$[
+  {
+      "partner": "Rohan Linguistics",
+      "project": "South Downs",
+      "mouStatus": "NotAttached",
```
Comment (CarsonF) [inline dbschema/seeds/017.partnerships.edgeql:6]: NotAttached is default for mouStatus and agreementStatus, and really I don't think it's used much. Let's just remove from these seeds.

File: dbschema/seeds/017.partnerships.edgeql [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+      "financialReportingType": "Funded",
```
Comment (CarsonF) [inline dbschema/seeds/017.partnerships.edgeql:7]: Only `Managing` type can set this field.
https://github.com/SeedCompany/cord-api-v3/blob/edgedb%2Fbump-gen-lib/src/components/partner/partner.service.ts#L126-L126

File: dbschema/seeds/017.partnerships.edgeql [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    select (
+      (select Partnership filter .partner = partner and .project = project) ??
+      (insert Partnership {
+        project := project,
+        partner := partner,
+        projectContext := project.projectContext,
+        mouStatus := <str>json_get(entry, 'mouStatus'),
+        financialReportingType := <str>json_get(entry, 'financialReportingType'),
+        agreementStatus := <str>json_get(entry, 'agreementStatus'),
+        primary := <bool>json_get(entry, 'primary'),
```
Comment (CarsonF) [inline dbschema/seeds/017.partnerships.edgeql:74]: Maybe we just assume primary true here?
```suggestion
        primary := <bool>json_get(entry, 'primary') ?? true,
```
And then only specify in json if false

File: dbschema/seeds/017.partnerships.edgeql [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+      "project": "South Downs",
+      "mouStatus": "NotAttached",
+      "financialReportingType": "Funded",
+      "agreementStatus": "NotAttached",
+      "primary": true,
+      "types": ["Funding"]
+  },
+  {
+      "partner": "Ered Luin Translation Syndicate",
+      "project": "Emyn Muil",
```
Comment (CarsonF) [inline dbschema/seeds/017.partnerships.edgeql:14]: Partnerships are typically thought about by `project` first. Would you mind putting those first here?
Then `partner`, then `types`.

**Review by CarsonF — COMMENTED** [review]
I think the data is close enough. Hard to see some of it without a UI to correlate, but we can iterate later if we need more code paths.

**Review by CarsonF — APPROVED** [review]
Thanks @brentkulwicki!

**Pull request merged**